inference/core/active_learning/configuration.py

```diff
@@ -60,76 +63,89 @@ def prepare_active_learning_configuration(
     )
     return initialise_active_learning_configuration(
         project_metadata=project_metadata,
+        model_id=model_id,
     )
 
 
 def prepare_active_learning_configuration_inplace(
-    api_key: str,
+    target_dataset_api_key: str,
+    target_dataset: str,
     model_id: str,
+    model_api_key: str,
     active_learning_configuration: Optional[dict],
 ) -> Optional[ActiveLearningConfiguration]:
     if (
         active_learning_configuration is None
         or active_learning_configuration.get("enabled", False) is False
     ):
         return None
-    dataset_id, version_id = get_model_id_chunks(model_id=model_id)
-    workspace_id = get_roboflow_workspace(api_key=api_key)
+    workspace_id = get_roboflow_workspace(api_key=target_dataset_api_key)
     dataset_type = get_roboflow_dataset_type(
-        api_key=api_key,
+        api_key=target_dataset_api_key,
         workspace_id=workspace_id,
-        dataset_id=dataset_id,
+        dataset_id=target_dataset,
     )
+    model_type = dataset_type
+    if not model_id.startswith(target_dataset):
+        model_type = get_model_type(model_id=model_id, api_key=model_api_key)
+    if model_type != dataset_type:
```

Comment:
Some types are compatible with each other (eg you should be able to store instance segmentation or keypoint detection preds into an object detection dataset).

Reply:
fair point - fixing